libs/application/templates/rental-agreement/src/fields/PropertySearch/index.tsx

```diff
@@ -130,14 +112,52 @@ export const PropertySearch: FC<React.PropsWithChildren<Props>> = ({
         {} as Record<string, boolean>,
       )
       setTableExpanded(expandedTables)
+
+      // Restore checked units based on storedValue.units
+      const initialCheckedUnits = storedValue.units.reduce(
+        (acc: Record<string, boolean>, unit: Unit) => {
+          const unitKey = `${unit.propertyCode}_${unit.unitCode}`
+          acc[unitKey] = unit.checked || false
+          return acc
+        },
+        {} as Record<string, boolean>,
+      )
+      setCheckedUnits(initialCheckedUnits)
+    }
+
+    // Restore numOfRooms per unit based on storedValue.units --> numOfRooms
+    if (storedValue?.units) {
+      const initialRoomsValue = storedValue.units.reduce(
+        (acc: Record<string, number>, unit: Unit) => {
+          const unitKey = `${unit.propertyCode}_${unit.unitCode}`
+          acc[unitKey] = unit.numOfRooms || 0
+          return acc
+        },
+        {} as Record<string, number>,
+      )
+      setNumOfRoomsValue(initialRoomsValue)
+    }
+    // Restore size value per unit based on storedValue.units --> changedSize | size
+    if (storedValue?.units) {
+      const initialSizeValue = storedValue.units.reduce(
+        (acc: Record<string, number>, unit: Unit) => {
+          const unitKey = `${unit.propertyCode}_${unit.unitCode}`
+          acc[unitKey] = unit.changedSize || 0
+          return acc
+        },
+        {} as Record<string, number>,
+      )
+      setUnitSizeChangedValue(initialSizeValue)
```

Comment:
Should this just run once, on page load? Just wondering if we should skip having storedValue in the dependency value? 
If this should run every time storedvalue changes then it is correct like this.